src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -2046,11 +2075,21 @@ export class SearchModel extends Disposable {
 		// In search on type case, delay the streaming of results just a bit, so that we don't flash the only "local results" fast path
 		this._startStreamDelay = new Promise(resolve => setTimeout(resolve, this.searchConfig.searchOnType ? 150 : 0));
 
-		const currentRequest = this.doSearch(query, progressEmitter, this._searchQuery, searchInstanceID, onProgress);
+		const req = this.doSearch(query, progressEmitter, this._searchQuery, searchInstanceID, onProgress, callerTokenSource);
```

Comment:
Webview results in notebooks needs an owner ID so that we can add and remove decorations properly

Reply:
I know it's not part of this PR but I'd think you could do that without exposing this webview thing all the way up at this layer

Reply:
The reason why it needs to be at the search model layer is because `FileMatches` that are part of the same search need to know the searchInstanceID to correctly remove webview decorations. Saying "I send a command from the search view to remove decorations from the webview" did not work, as there was a race condition with the new search starting to add its decorations before the old search instance began to dispose. I created https://github.com/microsoft/vscode/issues/191106 to track the polish of this work, as I agree it's strange to have this so far away from the webview work.